enterprise/causal-clustering/src/test/java/org/neo4j/causalclustering/helper/EnableableLifeCylcleEnableableStateChangeTest.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.causalclustering.helper;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.neo4j.causalclustering.helper.EnableableLifecycleStateTestHelpers.EnableableState;
+import org.neo4j.causalclustering.helper.EnableableLifecycleStateTestHelpers.LifeCycleState;
+import org.neo4j.logging.NullLogProvider;
+
+import static org.junit.Assert.assertEquals;
+import static org.neo4j.causalclustering.helper.EnableableLifecycleStateTestHelpers.setEnableableState;
+import static org.neo4j.causalclustering.helper.EnableableLifecycleStateTestHelpers.setInitialState;
+
+@RunWith( Parameterized.class )
+public class EnableableLifeCylcleEnableableStateChangeTest
```

Comment:
It's checking the enableable state on an enableable lifecycle. Makes sense to me, but is a mouthful.

Reply:
The name is correct but confusing since it is only operating on the enabled state. But i'll consider the naming overall. Got some suggestions further up.